typespec-extension/package.json

```diff
@@ -45,13 +45,13 @@
     "target/emitter.jar"
   ],
   "peerDependencies": {
-    "@typespec/compiler": ">=0.53.1 <1.0.0",
-    "@typespec/http": ">=0.53.0 <1.0.0",
-    "@typespec/rest": ">=0.53.0 <1.0.0",
-    "@typespec/openapi": ">=0.53.0 <1.0.0",
-    "@typespec/versioning": ">=0.53.0 <1.0.0",
-    "@azure-tools/typespec-azure-core": ">=0.40.0 <1.0.0",
-    "@azure-tools/typespec-client-generator-core": ">=0.40.0 <1.0.0"
+    "@typespec/compiler": ">=0.55.0 <1.0.0",
+    "@typespec/http": ">=0.55.0 <1.0.0",
+    "@typespec/rest": ">=0.55.0 <1.0.0",
+    "@typespec/openapi": ">=0.55.0 <1.0.0",
+    "@typespec/versioning": ">=0.55.0 <1.0.0",
+    "@azure-tools/typespec-azure-core": ">=0.41.0 <1.0.0",
+    "@azure-tools/typespec-client-generator-core": ">=0.41.2 <1.0.0"
```

Comment:
need TCGC 0.41.2 for bug fix within

that one probably need latest compiler etc., hence bump them as well.